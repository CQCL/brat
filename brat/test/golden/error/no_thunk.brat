ext "tket.X" X :: { Qubit -o Qubit }
ext "tket.Rz" Rz(Float) -> { Qubit -o Qubit }

-- Prior to https://github.com/CQCL-DEV/brat/pull/313,
-- the error message was that X didn't produce any (Braty) thunks
seq(n :: Nat) -> {Qubit -o Qubit}
seq(n) = {
    Rz(2.0 / n); X
}
