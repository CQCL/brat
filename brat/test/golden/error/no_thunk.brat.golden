Error in test/golden/error/no_thunk.brat on line 8:
    Rz(2.0 / n); X
             ^

  Type mismatch when checking n
Expected: (right :: Float)
But got:  (n :: Nat)


